Document Hierarchy Support on H2 (#1938)
cap-js/docs: advanced/odata.md

- <sup>(1)</sup> Beta feature, API may change

> ::: warning

Reviewer: ```suggestion
```

> Generic implementation is supported on the following databases

Reviewer 2: ```suggestion
Generic implementation is supported on the following databases:
```

|   | SAP HANA | H2 | PostgreSQL | SQLite |
|---|---|---|---|---|
| CAP Java | ✓ | ✓ | | |
| CAP Node.js | ✓ |  |✓ |✓ |

> :::

Reviewer: Does it still need to be a warning?

> :::

Reviewer: ```suggestion
```

- <sup>(1)</sup> Beta feature, API may change

::: warning

> Generic implementation is supported on SAP HANA (both Stacks), H2 (for Java) and Sqlite (for Node.js) Databases

Reviewer 2: ```suggestion
Generic implementation is supported on SAP HANA (both CAP stacks), H2 (CAP Java), Sqlite (CAP Node.js), and PostgreSQL (CAP Node.js) databases.
```

or

```suggestion
Generic implementation is supported on the following databases

|   | SAP HANA | H2 | PostgreSQL | SQLite |
|---|---|---|---|---|
| CAP Java | ✓ | ✓ | | |
| CAP Node.js | ✓ |  |✓ |✓ |
```
